solidity/contracts/bridge/Bridge.sol

```diff
@@ -132,30 +143,103 @@ contract Bridge {
         uint256 txIndexInBlock,
         bytes memory bitcoinHeaders
     ) external {
-        // TODO We need to read `fundingTxHash`, `fundingOutputIndex` and
-        //      P2SH script depositor address from `txInputVector`.
-        //      We then hash them to obtain deposit identifier and read
-        //      DepositInfo. From DepositInfo we know what amount was declared
-        //      by the depositor in their reveal transaction and we use that
-        //      amount to update their Bank balance, minus fee.
-        //
-        // TODO We need to validate if the sum in the output minus the
-        //      amount from the previous wallet balance input minus fees is
-        //      equal to the amount by which Bank balances were increased.
-        //
-        // TODO We need to validate txOutput to see if the balance was not
-        //      transferred away from the wallet before increasing balances in
-        //      the bank.
-        //
-        // TODO Delete deposit from unswept mapping or mark it as swept
-        //      depending on the gas costs. Alternativly, do not allow to
-        //      use the same TX input vector twice. Sweep should be provable
-        //      only one time.
+        // We need to read `fundingTxHash`, `fundingOutputIndex` and
+        // P2SH script depositor address from `txInputVector`.
+        // We then hash them to obtain deposit identifier and read
+        // DepositInfo. From DepositInfo we know what amount was declared
+        // by the depositor in their reveal transaction and we use that
+        // amount to update their Bank balance, minus fee.
+        require(txInputVector.validateVin(), "invalid input vector provided");
+        require(
+            //TODO: Change it, so that a single output can be handled
```

Comment:
Let's rather allow to pass a `txOutputVector` and just assert there is a single output. This way we won't need to change anything in Summa's lib. Also, if you pass a single out instead of entire vector, the tx hash will be wrong. The `txOutputVector` has some additional data about the outputs that must be included during hash computation.

Reply:
Done.